doc: continue in github (#260)
Innoactive/Creator: .Documentation/articles/innoactive-creator/default-conditions.md

### Description

This condition is fulfilled when the trainee grabs the `Object`. 
The condition is also fulfilled if the trainee already grabs the `Object` before the step is activated, so, if a trainee is already holding 
the specified object in hand while this condition is active, it is fulfilled immediately.

#### Interaction in VR

> To grab the object in VR, by default, the trainee has to move one of their hands/controllers into the object (its collider) and press the trigger button. The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.

Author: I see this topic as part of the referenced documentation coming afterwards.

Before the controller button is pressed, by default, the object is already highlighted to indicate that the trainee can now grab this object. The training designer can change this in the interactable highlighter script of the given object. You can enable/disable this interaction effect, change the highlight color and transparency for `On Touch`, `On Grab`, and `On Use`.

#### Interaction in Desktop Mode

To grab the object in Desktop Mode, the trainee has to move the mouse over the object (so that the ray hits the object's collider) and press the left mouse button.

> Before the mouse button is pressed, by default, the object is already highlighted to indicate that the trainee can now grab this object. This “on hover” effect can be changed by the template developer in the `Hover Highlight Handler` of the Desktop Mode Rig prefab. 

Author: ok

### Application Example

- Trainees need to grab objects to position them somewhere.
- Trainees need to grab a tool to use it.
- Trainees need to grab a component part to assemble it to another component part.

### Configuration

- #### Object

    The `Training Scene Object` to grab. The object needs to have the `Grabbable Property` and a collider component configured. The collider defines the area where the trainee can grab this object.

### Location of this Condition (for Developers)
This behavior is part of the <a href="https://github.com/Innoactive/Basic-Interaction-Component" target="_blank">Basic-Interaction</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Interaction-Component/blob/develop/Runtime/Conditions/GrabbedCondition.cs" target="_blank">here</a>.
 
------

## Move Object into Collider

### Description

This condition is fulfilled when the `Object` is within the specified `Collider` for the required amount of time (`Required seconds inside`) while this condition is active.

#### Interaction in VR

> In order to move the object into a collider, trainees need to move their hands/controllers into the object (its collider) and press the trigger button. While holding the trigger button and moving the arm, the object stays grabbed by trainees hands. Trainees move the object into the specified collider and release it. The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.

Reviewer: ```suggestion
In order to move the object into a collider, the trainee will do the following steps. 
1. Grab the object
2. Keep holding the button pressed to keep holding the object
3. Move towards the collider
4. Move the object into the specified collider and release it by releasing the grab button
The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.
```
We just explained grabbing before, let's not do that all over again :) Additionally, we probably should explain how to keep holding an object (by not releasing the grab button) in the grab condition instead of explaining it here. I think it makes more sense there.
And I suggest this with numbers here to make it easier to read.

Author: since users do not read this documentation from top to down, it is ok to be repetetitive.

#### Interaction in Desktop Mode
In order to move the object into a collider in Desktop Mode, trainees have to move the mouse on the object (so that the cast ray hit’s the objects collider) and press the left mouse button. The object is now grabbed, which is represented by an on-screen hand holding a miniature version of the object. In order to release the object into the collider, trainees move the mouse on the collider and press the lef mouse button. The on-screen hand disappears and the released object is now placed into the selected collider.

### Application Example
- You train air traffic controllers to give certain signs by waving red paddles. By positioning the collider above the trainee's head and one to the trainees right side, you can validate if the trainee raised one red paddle over his head and one to the side.

### Configuration

- #### Object

>     The Training Scene Object to grab. The object needs to have the `Grabbable Property` and a collider component configured. The collider defines the area where the trainee can grab this object.

Reviewer: Why is it here Training Scene Object but in grab condition it's the `Training Scene Object`. Just a small thing, but the format used should be the same in every condition to make it cohesive.

- #### Collider

>     The Training Scene Object with the collider inside which you want to move the Object to. Make sure that the Collider Object has a collider added and that the option `Is Trigger` is enabled.

Reviewer: ```suggestion
    The Training Scene Object with the collider you want to move the object to. Make sure that it has a collider added and that the option `Is Trigger` is enabled.
```

Author: done

- #### Required seconds inside

>     Set the time in seconds the Object should stay within the Collider.

Reviewer: ```suggestion
    Set the time in seconds that the object should stay inside the collider.
```

Author: done

### Location of this Condition (for Developers)

This behavior is part of the <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors" target="_blank">Basic-Conditions-And-Behaviors</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors/blob/develop/Runtime/Conditions/PositionalConditions/ObjectInColliderCondition.cs" target="_blank">here</a>.

------

## Object Nearby

### Description

This condition is fulfilled when the `Object` is within the specified `Range` of a `Reference object`.

#### Interaction in VR

> In order to move the object nearby a collider, trainees need to move their hands/controllers into the object (more precisely its collider) and press the trigger button. While holding the trigger button and moving the arm, the object stays attached to the trainees' hands. Trainees approach with the object the specified collider and release it. The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.

Reviewer: I don't understand what the specified collider is :o Maybe I am missing something?

> In order to move the object nearby a collider, trainees need to move their hands/controllers into the object (its collider) and press the trigger button. While holding the trigger button and moving the arm, the object stays attached to the trainees' hands. Trainees approach with the object the specified collider and release it. The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.

Author: people do not read top-down.

#### Interaction in Desktop Mode

> In order to approach the collider with an object in Desktop Mode, trainees have to move the mouse over the object (so that the cast ray hit’s the objects collider) and press the left mouse button. The object is now grabbed, which is represented by an on-screen hand holding a miniature version of the object. In order to release the object nearby the collider, trainees move the mouse over the collider and press the left mouse button. The on-screen hand disappears and the released object is now placed into the selected collider. `Object nearby` is difficult to simulate in Desktop Mode and is hence solved similar to `Move Object into collider`. 

Author: people do not read top-to-down

> In order to approach the collider with an object in Desktop Mode, trainees have to move the mouse over the object (so that the cast ray hit’s the objects collider) and press the left mouse button. The object is now grabbed, which is represented by an on-screen hand holding a miniature version of the object. In order to release the object nearby the collider, trainees move the mouse over the collider and press the left mouse button. The on-screen hand disappears and the released object is now placed into the selected collider. `Object nearby` is difficult to simulate in Desktop Mode and is hence solved similar to `Move Object into collider`. 

Author: we can add gifs in a next step

- #### Object

    The `Training Scene Object` that should be in the radius of the `Reference object`. Make sure you add at least the `Training Scene Object` component to this game object in the Unity Inspector. 
    
- #### Reference Object

   The `Training Scene Object` you want to measure the distance from.

- #### Range

    In this field, you can set the maximum distance between the *Object* and the *Reference object* required to fulfill this condition.The distance is calculated as the Euclidean norm between the transform’s positions of Object and Reference Object.

- #### Required seconds inside

    In this field, you can set the time in seconds the `Object` should stay within the radius `Range` of the `Reference object`.

### Location of this Condition (for Developers)

This behavior is part of the <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors" target="_blank">Basic-Conditions-And-Behaviors</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors/blob/develop/Runtime/Conditions/PositionalConditions/ObjectInRangeCondition.cs" target="_blank">here</a>.

------

## Release Object

### Description

> This condition is fulfilled when the `Object` is released by the trainee’s controller. If a trainee is already holding the specified object in hand while this condition is active, it is fulfilled immediately.

Reviewer: I still think this change or something similar needs to be added here. The condition release is not fulfilled immediately when the trainee is holding the object.

#### Snap Zone Generator
For any snappable object you can generate a snap zone that can snap exactly this object and makes it possible to use as a `Zone to snap into`. To do so, navigate to the `Snappable Property` in Unity's Inspector and click on the button `Create Snap Zone`. 

[![Snap Zone Generator](../images/default-conditions/snapzonegenerator.png "")](../images/default-conditions/snapzonegenerator.png)

#### Manual Snap Zone Creation
Instead of the automatic generation as described above, you can do those steps also manually. Please refer to available documention on the XRSocketInteractor from Unity or related sources. You can also make changes to the automatically created snap zone to adapt them to your needs. Please note, that these changes might impact the training process logic and can lead to breaking Creator logic. Do so on your own risk.

#### Feed Forward for Snap Zones

> Snap zones are restricted to which objects can be snapped to them, which means, any object can be valid (i.e. it can be snapped to this zone) or invalid (it can not be snapped to this zone) for a snap zone. In case you are moving a valid object into a zone (c.f. above, colliders and stuff), the snap zone color changes to ‘Validation Color’, giving the trainee a positive feedback. In case you are moving an invalid object into a zone, the snap zone color changes to ‘Invalid Color’, giving the trainee the feedback that this is the wrong object for this zone. 

Author: done

Which colors and which materials are to be used can be changed in the Snap Zones parameters and settings.

#### Snap Zone Parameters and Settings
To change the highlight color or validation hover material of a dedicated snap zone, navigate to the snap zone object in the Unity inspector. In the script `Snap Zone` you will find these parameters among others. 

[![Snap Zone Parameters](../images/default-conditions/snapzoneparameters.png "")](../images/default-conditions/snapzoneparameters.png)

To change the colors and materials of all snap zones in the scene, select them in the Creator snap zone settings and press 'Apply settings in current scene'.

[![Snap Zone Settings](../images/default-conditions/snapzonesettings.png "")](../images/default-conditions/snapzonesettings.png)

The snap zone settings can be found in tab “Innoactive” -> “Settings” -> “Snap Zones”.

### Application Example

- Teach trainees where to position certain component parts in a car.

### Configuration

- #### Object

>     The Training Scene Object to place (snap). The object needs to have the `Grabbable Property` and a collider component configured. 

Author: done

 
 

------

## Use Object

### Description

This condition is fulfilled when the `Object` is used by pressing the *Use* button of the controller while being touched or grabbed.

#### Interaction in VR
To use an object in VR, by default, trainees have to move one of their hands/controllers into the object (its collider) and press the trigger button. The button to trigger can be changed by the template developer as described <a href="https://developers.innoactive.de/documentation/creator/latest/articles/xr/index.html" target="_blank">here</a>.

#### Interaction in Desktop Mode
In order to use an object in desktop mode, it needs to be grabbed before (see [Grab Condition](https://developers.innoactive.de/documentation/creator/latest/articles/innoactive-creator/default-conditions.html#grab-object)). A grabbed object is displayed in miniature size inside an on-screen hand. Trainees click the button next to it, titled *use object*, to use the previously selected object.

### Application Example
- The trainee picks up a drill and drills in screws.

> - The trainee pushes a button on a machine.

Reviewer: Isn't this a touch example?

### Description

This condition is fulfilled when the trainee grabs the `Object`. 
The condition is also fulfilled if the trainee already grabs the `Object` before the step is activated, so, if a trainee is already holding 
the specified object in hand while this condition is active, it is fulfilled immediately.

### Interaction in VR

> To grab the object in VR, by default, the trainee has to move one of their hands/controllers into the object (it’s collider) and press the trigger button. The button to trigger can be changed by the template developer as described here (TO BE DONE).

Reviewer 2: Dont forget the "TO BE DONE"

### Interaction in Desktop Mode

> To grab the object in Desktop Mode, the trainee has to move the mouse on the object (so that the cast ray hit’s the objects collider) and press the left mouse button. The button to trigger can be changed by the template developer as described here (TO BE DONE).

Reviewer 2: Not sure if the "raycast sentence" in parentheses is needed. very technical description

Dont forget the "TO BE DONE"

### Interaction Highlights
Before the controller button or, respectively, mouse button is pressed, by default, the object is already highlighted to indicate that the trainee can now grab this object. This “on hover” effect can be changed by the template developer here (TO BE DONE). The training designer can also change this in the interactable highlighter property of the given object as described here. You can easily enable/disable this hover effect, or change the highlight color and transparency (TO BE DONE).

### Application Example

- Trainees need to grab objects to position them somewhere.
- Trainees need to grab a tool to use it.
- Trainees need to grab a component part to assemble it to another component part.

### Configuration

- #### Object

    The `Training Scene Object` to grab. The object needs to have the `Grabbable Property` and a collider component configured. The collider defines the area where the trainee can grab this object.

### Location of this Condition (for Developers)

> This behavior is part of the <a href="https://github.com/Innoactive/Basic-Interaction-Component" target="_blank">Basic-Interaction</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Interaction-Component/blob/develop/Runtime/Conditions/GrabbedCondition.cs" target="_blank">here</a>

Reviewer 2: this is not how you do links in md 🤔 

`[Basic-Interaction](https://github.com/Innoactive/Basic-Interaction-Component)`

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

 
------

## Move Object into Collider
### Description

This condition is fulfilled when the `Object` is within the specified `Collider` for the required amount of time (`Required seconds inside`) while this condition is active.

### Interaction in VR
### Interaction in Desktop Mode
### Interaction Highlights

### Application Example

> - You train air traffic controllers to give certain signs by waving red paddles. By positioning the collider above the trainee's head and one to the trainees right side, you can validate if the trainee raised one red paddle over his head and one to the side.

Reviewer 3: We have this example in 'Object Nearby' condition. Can be moved here since a collider is probably better for defining the safety/emergency zone.

> - You train air traffic controllers to give certain signs by waving red paddles. By positioning the collider above the trainee's head and one to the trainees right side, you can validate if the trainee raised one red paddle over his head and one to the side.

Reviewer: I agree with Thomas here :) I think this example is better suited for this condition than for nearby in general as well.

- #### Collider

    The Training Scene Object with the collider inside which you want to move the Object to. Make sure that the Collider Object has a collider added and that the option ‘Is Trigger’ is enabled.

- #### Required seconds inside

    Set the time in seconds the Object should stay within the Collider.

### Location of this Condition (for Developers)

This behavior is part of the <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors" target="_blank">Basic-Conditions-And-Behaviors</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors/blob/develop/Runtime/Conditions/PositionalConditions/ObjectInColliderCondition.cs" target="_blank">here</a>  

------

## Object Nearby

### Description

This condition is fulfilled when the `Object` is within the specified `Range` of a `Reference object`.

### Interaction in VR
### Interaction in Desktop Mode
### Interaction Highlights

### Application Example

- The trainee is supposed to hold a RFID tag included in his gloves (represented by his controller) near a scanner.

> - If a trainee moves to close to a machinery while it is operating, the emergency shut down is triggered.

Reviewer: agreeing with this

### Configuration

- #### Object

>     The `Training Scene Object` that should be in the radius of the `Reference object`. Make sure you add at least the `Training Scene Object` component to this game object in the Unity Inspector. 

Reviewer 2: ```suggestion
    The `Training Scene Object` that should be in the radius of the `Reference Object`. Make sure you add at least the `Training Scene Object` component to this game object in the Unity Inspector. 
```

Reviewer: also agree with this

    
- #### Reference Object

   The `Training Scene Object` you want to measure the distance from.

- #### Range

>     In this field, you can set the maximum distance between the *Object* and the *Reference object* required to fulfill this condition.The distance is calculated as the Euclidean norm between the transform’s positions of Object and Reference Object.

Reviewer 2: Before you always used code tags for object and reference object, so keep it consistent

```suggestion
    In this field, you can set the maximum distance between the `Object` and the `Reference Object` required to fulfill this condition. The distance is calculated as the Euclidean norm between the transform’s positions of `Object` and `Reference Object`.
```

- #### Required seconds inside

    In this field, you can set the time in seconds the `Object` should stay within the radius `Range` of the `Reference object`.

### Location of this Condition (for Developers)

This behavior is part of the <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors" target="_blank">Basic-Conditions-And-Behaviors</a> component. The file is located <a href="https://github.com/Innoactive/Basic-Conditions-And-Behaviors/blob/develop/Runtime/Conditions/PositionalConditions/ObjectInRangeCondition.cs" target="_blank">here</a>  

------

## Release Object

### Description

This condition is fulfilled when the `Object` is released by the trainee’s controller. If a trainee is already holding the specified object in hand while this condition is active, it is fulfilled immediately.
### Interaction in VR
### Interaction in Desktop Mode
### Interaction Highlights

### Application Example

### Configuration

- #### Object

>     the `Training Scene Object` to release. The object needs to have the Grabbable Property and a collider component configured. 

Reviewer 2: ```suggestion
    The `Training Scene Object` to release. The object needs to have the Grabbable Property and a collider component configured. 
```
